src/compiler/checker.ts

```diff
@@ -6171,10 +6171,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 const overrideTypeNode = typeViaParent && jsDocProperty.typeExpression && getTypeFromTypeNode(context, jsDocProperty.typeExpression.type) !== typeViaParent ? typeToTypeNodeHelper(typeViaParent, context) : undefined;
                 return overrideTypeNode;
             },
-            enterNewScope(context, node) {
+            enterNewScope(context, node, expandParameters = true) {
                 if (isFunctionLike(node) || isJSDocSignature(node)) {
                     const signature = getSignatureFromDeclaration(node);
-                    const expandedParams = getExpandedParameters(signature, /*skipUnionExpanding*/ true)[0];
+                    const expandedParams = !expandParameters ? signature.parameters : getExpandedParameters(signature, /*skipUnionExpanding*/ true)[0];

```

Comment:
Believe it or not, all tests pass if we just unconditionally use `signature.parameters` here (not expanding at all). I'm not an expert in why we needed to expand them before, but it sure feels like we don't need to do them at all anymore?

Reply:
You are right. Expansion no longer ever happens from the syntactic printer. Only in the type checker printer. Removed the option